common/tasks/interleaved_weighted_round_robin_test.go

```diff
@@ -86,7 +86,7 @@ func (s *interleavedWeightedRoundRobinSchedulerSuite) SetupTest() {
 		2: 2,
 		3: 1,
 	}
-	s.channelWeightUpdateCh = make(chan struct{})
+	s.channelWeightUpdateCh = make(chan struct{}, 1)
```

Comment:
Prod code don't have this issue
https://github.com/temporalio/temporal/blob/master/service/history/queues/scheduler.go#L140